x-pack/legacy/plugins/infra/public/pages/logs/analysis/setup/analysis_setup_indices_form.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo, useCallback } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { EuiCode, EuiDescribedFormGroup, EuiFormRow, EuiCheckboxGroup } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+const indicesSelectionLabel = i18n.translate('xpack.infra.analysisSetup.indicesSelectionLabel', {
+  defaultMessage: 'Indices',
+});
+
+type IndicesSelection = Record<string, boolean>;
+
+export const AnalysisSetupIndicesForm: React.FunctionComponent<{
+  indices: IndicesSelection;
+  onChangeSelectedIndices: (selectedIndices: IndicesSelection) => void;
+  validationErrors?: Array<'TOO_FEW_SELECTED_INDICES'>;
```

Comment:
Would it make sense to reuse `ValidationErrors[]` here?

Reply:
Introducing an `import´ dependency on the container at this level seems wrong. I'll try to see if I can factor it out somewhere.